ruff_cli/src/printer.rs

```diff
@@ -336,6 +343,56 @@ impl<'a> Printer<'a> {
         Ok(())
     }
 
+    pub fn write_statistics(&self, diagnostics: &Diagnostics) -> Result<()> {
+        let mut violations = diagnostics
+            .messages
+            .iter()
+            .map(|message| message.kind.rule())
+            .collect::<Vec<_>>();
+        violations.sort();
+        violations.dedup();
+        let statistics = violations.iter().map(|rule| ExpandedStatistics {
+            code: rule.code(),
+            count: diagnostics
+                .messages
+                .iter()
+                .filter(|message| message.kind.rule() == *rule)
+                .count(),
+            message: diagnostics
+                .messages
+                .iter()
+                .find(|message| message.kind.rule() == *rule)
+                .map(|message| message.kind.body())
+                .unwrap(),
+        });
+
+        let mut stdout = BufWriter::new(io::stdout().lock());
+        match self.format {
+            SerializationFormat::Json => {
+                writeln!(
+                    stdout,
+                    "{}",
+                    serde_json::to_string_pretty(&statistics.collect::<Vec<_>>())?
+                )?;
+            }
+            SerializationFormat::Junit => {}
+            SerializationFormat::Text => {
+                for msg in statistics {
+                    writeln!(stdout, "{:<5} {} {}", msg.count, msg.code, msg.message)?;
+                }
```

Comment:
It would be nice to add a comment that this uses the same output format as `flake8 --statistics`.

Reply:
added a comment

Reply:
nit: it would be nice to specifically mention `flake8 --statistics` instead of just flake8

Reply:
done